code/extensions/Lumberjack.php

```diff
@@ -1,150 +1,167 @@
+
+		if($this->shouldFilter()) {
+			// Filter the SiteTree
+			return $staged->exclude("ClassName", $this->owner->getExcludedSiteTreeClassNames());
+		}
+		return $staged;
+	}
+
+
+	/**
+	 * Augments (@link Hierarchy::liveChildren()} by hiding excluded child classnames
+	 *
+	 * @param boolean $showAll Include all of the elements, even those not shown in the menus.
+	 *   (only applicable when extension is applied to {@link SiteTree}).
+	 * @param boolean $onlyDeletedFromStage Only return items that have been deleted from stage
+	 * @return SS_List
+	 */
+	public function liveChildren($showAll = false, $onlyDeletedFromStage = false) {
+		$baseClass = ClassInfo::baseDataClass($this->owner->class);
+		$children = $baseClass::get()
+			->filter('ParentID', (int)$this->owner->ID)
+			->exclude('ID', (int)$this->owner->ID)
+			->setDataQueryParam(array(
+				'Versioned.mode' => $onlyDeletedFromStage ? 'stage_unique' : 'stage',
+				'Versioned.stage' => 'Live'
+			));
+
+		if(!$showAll) {
+			$children = $children->filter('ShowInMenus', 1);
+		}
+
+		if($this->shouldFilter()) {
+			// Filter the SiteTree
+			return $children->exclude("ClassName", $this->owner->getExcludedSiteTreeClassNames());
+		}
+		return $children;
+	}
+
+
+	/**
+	 * This returns the title for the tab and GridField. This can be overwritten
+	 * in the owner class.
+	 *
+	 * @return string
+	 */
+	protected function getLumberjackTitle() {
+		if(method_exists($this->owner, 'getLumberjackTitle')) {
+			return $this->owner->getLumberjackTitle();
+		}
+		return _t("Lumberjack.TabTitle", "Child Pages");
+	}
+
+
+	/**
+	 * This returns the gird field config for the lumberjack gridfield.
+	 *
+	 * @return GridFieldConfig
+	 */
+	protected function getLumberjackGridFieldConfig() {
+		if(method_exists($this->owner, 'getLumberjackGridFieldConfig')) {
+			return $this->owner->getLumberjackGridFieldConfig();
+		}
+		return GridFieldConfig_Lumberjack::create();
+	}
+
+
+	/**
+	 * Checks if we're on a controller where we should filter. ie. Are we loading the SiteTree?
+	 *
+	 * @return bool
+	 */
+	protected function shouldFilter() {
+		$controller = Controller::curr();
+		return $controller instanceof LeftAndMain
+			&& in_array($controller->getAction(), array("treeview", "listview", "getsubtree"));
+	}
+
+
+	/**
+     * Returns list of pages for the CMS gridfield
+     *
+     * This also allows the owner class to override this method, e.g. to provide custom ordering.
+     *
+     * @var array $excluded     List of class names excluded from the SiteTree
+     * @return DataList
+     */
+	public function getLumberjackPagesForGridfield($excluded = array()) {
+        if (method_exists($this->owner, 'getLumberjackPagesForGridfield')) {
+            return $this->owner->getLumberjackPagesForGridfield($excluded);
+        }
+
+        return SiteTree::get()->filter(array(
+            'ParentID' => $this->owner->ID,
+            'ClassName' => $excluded,
```

Comment:
I was a bit torn here as well, but then I stuck with the convention used, not to rename variables all over the class.